x/staking/legacy/v040/migrate.go

```diff
@@ -42,9 +42,14 @@ func Migrate(stakingState v038staking.GenesisState) *v040staking.GenesisState {
 
 	newValidators := make([]v040staking.Validator, len(stakingState.Validators))
 	for i, oldValidator := range stakingState.Validators {
+		conPubKeyAny, err := codectypes.PackAny(oldValidator.ConsPubKey)
```

Comment:
`oldValidator.ConsPubKey` is a `crypto.PubKey`, which is a tmed25519.PubKey. we might need a conversion to our own ed25519 here.